src/librustc_mir/build/matches/mod.rs

```diff
@@ -188,8 +189,17 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                     self.user_assert_ty(block, ty, var, irrefutable_pat.span);
                 }
 
-                self.schedule_drop_for_binding(var, irrefutable_pat.span, OutsideGuard);
-                self.into(&place, block, initializer)
+                let block = self.into(&place, block, initializer);
```

Comment:
This needs to be in between the storage and the value drop.

Reply:
Right, I remember I was playing around in the end.